Installing-Necessary-CLI-Tools.md

```diff
@@ -0,0 +1,35 @@
+# Installing CLI Tools Developers Should Not Live Without (Linux and Mac)
+
+This article is a short description on how to install key command line utilities that developers use everyday on Macintosh and Linux environments. The main CLI tools that will be shown are: Homebrew (Mac), Node and NPM, Bower, Gulp and Grunt.
+
+### Installing Homebrew (Macintosh Systems)
+
+Homebrew is ‘The missing package manager for OS X’. It is a great tool for downloading and installing packages straight from your command line. This is not needed on Linux distributions because they already have package managers installed by default such as `apt-get` or `pacman`. To install Homebrew, simply paste the following command in your terminal:
+* `/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"`
+
+### Installing NPM
+
+`NPM`, or Node Package Manager, is another useful package manager for downloading mostly web tools. Downloading `NPM` will also install the Node.js framework.
+
+##### Mac:
+* Using `Homebrew` type: `brew install node` and `NPM` should have been installed.
+
+##### Linux:
+* Using `apt-get` first type: `curl -sL https://deb.nodesource.com/setup_4.x | sudo -E bash -`, then `sudo apt-get install nodejs`
+* Using `pacman` type: `pacman -S nodejs npm`
+* Using `yum` type: `sudo yum install nodejs npm`
+
+### Installing Bower
+Bower is a popular package manager that allows you to install front-end libraries. You can install it on Linux and Macintosh systems using `npm` with the following command:
+* `npm install bower`
+(Same command for both Linux and OS X)
+
+### Installing Gulp
```

Comment:
It would be a good idea to mention it to the reader that Gulp and Grunt both are task runners, but one focuses on config while other focuses on code. 

So if they have one, they might not need the other.